File: frontend/src/lib/types/admin-general-settings.ts
// Mirrors backend/app/schemas/admin_general_settings.py exactly — keep in sync.

export interface GeneralSettingsSchema {
  site_name: string;
  contact_email: string | null;
  data_access_email: string | null;
  max_upload_size_mb: number;
  session_lifetime_min: number;
}

export interface GeneralSettingsUpdate {
  site_name?: string;
  contact_email?: string | null;
  data_access_email?: string | null;
  max_upload_size_mb?: number;
  session_lifetime_min?: number;
}

export interface NotificationSettingsSchema {
  notify_new_request: boolean;
  notify_new_user: boolean;
  notify_expiring_dataset: boolean;
}

export interface NotificationSettingsUpdate {
  notify_new_request?: boolean;
  notify_new_user?: boolean;
  notify_expiring_dataset?: boolean;
}

// Mirrors backend/app/schemas/admin_overview.py's storage-capacity pair.
export interface StorageCapacitySchema {
  storage_capacity_bytes: number | null;
}

export interface StorageCapacityUpdate {
  storage_capacity_bytes: number | null;
}

File: backend/app/schemas/admin_general_settings.py
from pydantic import BaseModel, Field


class GeneralSettingsSchema(BaseModel):
    site_name: str
    contact_email: str | None
    data_access_email: str | None
    max_upload_size_mb: int
    session_lifetime_min: int

    model_config = {"from_attributes": True}


class GeneralSettingsUpdate(BaseModel):
    site_name: str | None = Field(default=None, min_length=1, max_length=255)
    contact_email: str | None = None
    data_access_email: str | None = None
    max_upload_size_mb: int | None = Field(default=None, gt=0)
    session_lifetime_min: int | None = Field(default=None, gt=0)


class NotificationSettingsSchema(BaseModel):
    notify_new_request: bool
    notify_new_user: bool
    notify_expiring_dataset: bool

    model_config = {"from_attributes": True}


class NotificationSettingsUpdate(BaseModel):
    notify_new_request: bool | None = None
    notify_new_user: bool | None = None
    notify_expiring_dataset: bool | None = None
